BSP/Simulator/Src/Simulator.c

```diff
@@ -180,18 +178,20 @@ static void readInputFile(char *jsonPath) {
                 tail->temperatures[idx] = cJSON_GetArrayItem(temperatureArray, idx)->valueint;
             }
         }
-        if (!current) { // current in mA
+        if (!currentObj) { // current in mA
             printf("Current simulator state does not have a specified current value. Using 30...\n");
             tail->current = 30;
+        } else {
+            tail->current = currentObj->valueint;
```

Comment:
this is the fix for current and charge not being set from the json (I forgot to set their values when rewriting the cjson parse stuff)